Apps/GatewayApi/test/unit/Services.Test/UserFeedbackServiceTests.cs

```diff
@@ -105,84 +178,43 @@ private static Mock<ICryptoDelegate> GetCryptoDelegateMock()
+            Mock<IAuthenticationDelegate>? authenticationDelegateMock = null)
         {
-            const string dbErrorMessage = "DB Error!";
-            Guid ratingId = Guid.NewGuid();
-
-            SubmitRating createRating = new()
-            {
-                RatingValue = 5,
-                Skip = false,
-            };
+            feedbackDelegateMock ??= new();
+            ratingDelegateMock ??= new();
+            userProfileDelegateMock ??= new();
+            backgroundJobClientMock ??= new();
+            authenticationDelegateMock ??= new();
 
-            Rating rating = new()
-            {
-                Id = ratingId,
-                RatingValue = createRating.RatingValue,
-                Skip = createRating.Skip,
-            };
-
-            DbResult<Rating> dbResult = new()
-            {
-                Payload = rating,
-                Status = dbErrorExists ? DbStatusCode.Error : DbStatusCode.Created,
-                Message = dbErrorExists ? dbErrorMessage : string.Empty,
-            };
-
-            RequestResult<RatingModel> expected = new()
-            {
-                ResultStatus = dbErrorExists ? ResultType.Error : ResultType.Success,
-                ResultError = dbErrorExists
-                    ? new()
-                    {
-                        ResultMessage = dbErrorMessage,
-                        ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationInternal, ServiceType.Database),
-                    }
-                    : null,
-                ResourcePayload = dbErrorExists ? null : new() { Id = ratingId, RatingValue = rating.RatingValue, Skip = rating.Skip },
-            };
-
-            Mock<IRatingDelegate> ratingDelegateMock = new();
-            ratingDelegateMock.Setup(s => s.InsertRatingAsync(It.Is<Rating>(r => r.RatingValue == rating.RatingValue && r.Skip == rating.Skip), It.IsAny<CancellationToken>()))
-                .ReturnsAsync(dbResult);
-
-            IUserFeedbackService service = new UserFeedbackService(
+            return new UserFeedbackService(
                 new Mock<ILogger<UserFeedbackService>>().Object,
-                new Mock<IFeedbackDelegate>().Object,
+                feedbackDelegateMock.Object,
                 ratingDelegateMock.Object,
-                new Mock<IUserProfileDelegate>().Object,
-                new Mock<IBackgroundJobClient>().Object,
+                userProfileDelegateMock.Object,
+                backgroundJobClientMock.Object,
                 MappingService,
-                new Mock<IAuthenticationDelegate>().Object);
-
-            return new(service, expected, createRating);
+                authenticationDelegateMock.Object);
         }
 
-        private static CreateUserFeedbackMock SetupCreateUserFeedbackMock()
+        private static IUserFeedbackService SetupCreateRatingMock(DbResult<Rating> dbResult, SubmitRating createRating)
```

Comment:
Method name no longer corresponds to the returned type.  I suggest something like `SetupUserFeedbackServiceForCreateRating()`.

Reply:
done